src/widgets/helper/ChannelView.cpp

```diff
@@ -601,6 +606,24 @@ void ChannelView::themeChangedEvent()
     this->setupHighlightAnimationColors();
     this->queueLayout();
     this->messageColors_.applyTheme(getTheme());
+    if (this->colorVisitor_)
+    {
+        this->colorVisitor_(this->messageColors_, getTheme());
+    }
```

Comment:
Could `applyTheme` on MessageColors accept the `overlayBackgroundOpacity` plus a boolean whether this is an overlay enough, and let it do the logic for how to apply the colors instead?

so `MessageColors::applyTheme(Theme*)` would instead be `MessageColors::applyTheme(Theme*, bool isInOverlay, int backgroundOpacity`) - that should let us get rid of the visitor pattern